src/pdf_beamtime/include/pdf_beamtime/simple_server.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef SIMPLE_SERVER_HPP
+#define SIMPLE_SERVER_HPP
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <functional>
+#include <memory>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+#include <hello_moveit_interfaces/action/simple_action_msg.hpp>
+
+/// @brief Create the obstacle environment and an simple action server for the robot to move
+class SimpleServer
+{
+public:
+  using SimpleActionMsg = hello_moveit_interfaces::action::SimpleActionMsg;
+
+  SimpleServer(const std::string & move_group_name, const rclcpp::NodeOptions & options);
+  rclcpp::node_interfaces::NodeBaseInterface::SharedPtr getNodeBaseInterface();
+
+private:
+  rclcpp::Node::SharedPtr node_;
+
+  /// @brief Pointer to the action server
+  rclcpp_action::Server<SimpleActionMsg>::SharedPtr action_server_;
+
+  const std::string ACTION_NAME = "pdf_simple_action";
+
+  moveit::planning_interface::PlanningSceneInterface * planning_scene_interface;
+  /// @brief records the two types of obstacles CYLINDER and BOX.
+  std::map<std::string, int> obstacle_type_map;
+
+  moveit::planning_interface::MoveGroupInterface * move_group_interface_;
```

Comment:
Does this need to be a pointer? 
https://github.com/maffettone/erobs/blob/07e61ea4365613a071ecead56364827a07ff505b/src/hello_moveit/src/pick_place_repeat_server.cpp#L88C3-L88C49

Reply:
https://github.com/ChandimaFernando/erobs/blob/9f696b7d2d205fd2038f6a764db896805352b3fb/src/pdf_beamtime/include/pdf_beamtime/pdf_beamtime_server.hpp#L39

Pointer is removed